library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/ExtensionSupport.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay
+
+import android.view.View
+import com.datadog.android.sessionreplay.internal.recorder.mapper.WireframeMapper
+
+/**
+ * In case you need to provide different configuration for a specific Android UI framework that
+ * is not supported by our SR instrumentation layer (e.g Material elements)
+ * you can implement this class.
+ * @see [SessionReplayConfiguration.Builder.addExtensionSupport]
+ */
+interface ExtensionSupport {
+    /**
+     * Use this method if you want to apply a custom [WireframeMapper] for a specific [View] type.
+     * @return the [View] type to custom [WireframeMapper] as a [Map].
+     */
+    fun getCustomViewMappers(): Map<Class<*>, WireframeMapper<View, *>>
```

Comment:
I'm missing some context, but why the signature of `WireframeMapper` here is like that, while at the declaration site it has some boundaries set `<T : View, S : MobileSegment.Wireframe>`?

Reply:
It is hard to explain so I will explain you at the desk :). In a nutshell it is because those boundaries are set by default and allows also wildcard. Basically in Kotlin `WireframeMapper<View,*>` relates to `WireframeMapper<View, T:Wireframe>`. The idea is that I prefer to cast my mappers here where I provide than later. So if I would provide a `Map<Class<T>, WireframeMapper<T,S>` from here later I would have to cast it to `Map<Class<*>, WireframeMapper<View, *>>` in order to use it in my `GenericMapper`. I'd rather provided it directly like that and do the casting in the provider method.